Build trouble on FreeBSD

Hello! Sorry to bother but I'm getting the following error building under FreeBSD 13.0 with the latest master and really have no idea how to fix it. All listed requirements installed and followed the build instructions (hopefully) precisely. Thank you already.

> [libstdc++/build/kernel] checking for i686-pc-serenity-ranlib... i686-pc-serenity-ranlib
[libstdc++/build/kernel] checking for i686-pc-serenity-strip... i686-pc-serenity-strip
[libstdc++/build/kernel] checking whether ln -s works... yes
[libstdc++/build/kernel] checking for i686-pc-serenity-gcc...  /data/source/serenity/Toolchain/Build/i686/gcc/./gcc/xgcc -B/data/source/serenity/Toolchain/Build/i686/gcc/./gcc/ -B/data/source/serenity/Toolchain/Local/i686/i686-pc-serenity/bin/ -B/data/source/serenity/Toolchain/Local/i686/i686-pc-serenity/lib/ -isystem /data/source/serenity/Toolchain/Local/i686/i686-pc-serenity/include -isystem /data/source/serenity/Toolchain/Local/i686/i686-pc-serenity/sys-include
[libstdc++/build/kernel] checking for suffix of object files... configure: error: in `/data/source/serenity/Toolchain/Build/i686/gcc/i686-pc-serenity/libgcc':
[libstdc++/build/kernel] configure: error: cannot compute suffix of object files: cannot compile
[libstdc++/build/kernel] See `config.log' for more details
[libstdc++/build/kernel] gmake: *** [Makefile:13430: configure-target-libgcc] Error 1
XXX install libstdc++
[libstdc++/install/kernel] /bin/sh /data/source/serenity/Toolchain/Tarballs/gcc-11.1.0/mkinstalldirs /data/source/serenity/Build/i686/Temp/data/source/serenity/Toolchain/Local/i686 /data/source/serenity/Build/i686/Temp/data/source/serenity/Toolchain/Local/i686
[libstdc++/install/kernel] mkdir /data/source/serenity/Build/i686/Temp
[libstdc++/install/kernel] mkdir /data/source/serenity/Build/i686/Temp/data
[libstdc++/install/kernel] mkdir /data/source/serenity/Build/i686/Temp/data/source
[libstdc++/install/kernel] mkdir /data/source/serenity/Build/i686/Temp/data/source/serenity
[libstdc++/install/kernel] mkdir /data/source/serenity/Build/i686/Temp/data/source/serenity/Toolchain
[libstdc++/install/kernel] mkdir /data/source/serenity/Build/i686/Temp/data/source/serenity/Toolchain/Local
[libstdc++/install/kernel] mkdir /data/source/serenity/Build/i686/Temp/data/source/serenity/Toolchain/Local/i686
[libstdc++/install/kernel] gmake[1]: Entering directory '/data/source/serenity/Toolchain/Build/i686/gcc/i686-pc-serenity/libgcc'
[libstdc++/install/kernel] gmake[1]: *** No rule to make target 'install'.  Stop.
[libstdc++/install/kernel] gmake[1]: Leaving directory '/data/source/serenity/Toolchain/Build/i686/gcc/i686-pc-serenity/libgcc'
[libstdc++/install/kernel] gmake: *** [Makefile:13583: install-target-libgcc] Error 2
cp: /data/source/serenity/Build/i686/Temp//data/source/serenity/Toolchain/Local/i686/*: No such file or directory

--- a/Ports/binutils/patches/binutils.patch
+++ b/Ports/binutils/patches/binutils.patch
@@ -133,3 +133,15 @@ diff -Naur binutils-2.36.1/ld/Makefile.in binutils-2.36.1.serenity/ld/Makefile.i
  	ehppa64linux.c \
  	ei386pep.c \
  	emmo.c
+diff -Naur binutils-2.36.1/binutils/objcopy.c binutils-2.36.1.serenity/binutils/objcopy.c
+--- binutils-2.36.1/binutils/objcopy.c	2021-02-05 15:13:40.000000000 +0100
++++ binutils-2.36.1.serenity/binutils/objcopy.c	2021-06-07 08:28:08.961581287 +0200
+@@ -3769,7 +3769,7 @@
+   /* To allow us to do "strip *" without dying on the first
+      non-object file, failures are nonfatal.  */
+   ibfd = bfd_openr (input_filename, input_target);
+-  if (ibfd == NULL || fstat (fileno (ibfd->iostream), in_stat) != 0)
++  if (ibfd == NULL || bfd_stat (ibfd, in_stat) != 0)
+     {
+       bfd_nonfatal_message (input_filename, NULL, NULL, NULL);
+       status = 1;

--- a/Toolchain/Patches/binutils.patch
+++ b/Toolchain/Patches/binutils.patch
@@ -133,3 +133,15 @@ diff -Naur binutils-2.36.1/ld/Makefile.in binutils-2.36.1.serenity/ld/Makefile.i
  	ehppa64linux.c \
  	ei386pep.c \
  	emmo.c
+diff -Naur binutils-2.36.1/binutils/objcopy.c binutils-2.36.1.serenity/binutils/objcopy.c
+--- binutils-2.36.1/binutils/objcopy.c	2021-02-05 15:13:40.000000000 +0100
++++ binutils-2.36.1.serenity/binutils/objcopy.c	2021-06-07 08:28:08.961581287 +0200
+@@ -3769,7 +3769,7 @@
+   /* To allow us to do "strip *" without dying on the first
+      non-object file, failures are nonfatal.  */
+   ibfd = bfd_openr (input_filename, input_target);
+-  if (ibfd == NULL || fstat (fileno (ibfd->iostream), in_stat) != 0)
++  if (ibfd == NULL || bfd_stat (ibfd, in_stat) != 0)
+     {
+       bfd_nonfatal_message (input_filename, NULL, NULL, NULL);
+       status = 1;

